dvc/repo/worktree.py

```diff
@@ -65,91 +110,115 @@ def outs_filter(out: "Output") -> bool:
-        push=True,
-        targets=targets,
-        **kwargs,
-    )
-    new_index = view.data["repo"]
-    if remote.worktree:
-        logger.debug("indexing latest worktree for '%s'", remote.path)
-        old_index = build_data_index(view, remote.path, remote.fs)
-        logger.debug("Pushing worktree changes to '%s'", remote.path)
-    else:
-        old_index = None
-        logger.debug("Pushing version-aware files to '%s'", remote.path)
-
-    if remote.worktree:
-        diff_kwargs: Dict[str, Any] = {
-            "meta_only": True,
-            "meta_cmp_key": partial(_meta_checksum, remote.fs),
-        }
-    else:
-        diff_kwargs = {}
+    pushed = 0
+    stages: Set["Stage"] = set()
 
-    total = len(new_index)
-    with Callback.as_tqdm_callback(
-        unit="file", desc="Push", disable=total == 0
-    ) as cb:
-        cb.set_size(total)
-        pushed = checkout(
-            new_index,
-            remote.path,
-            remote.fs,
-            old=old_index,
-            delete=remote.worktree,
-            callback=cb,
-            latest_only=remote.worktree,
-            jobs=jobs,
-            **diff_kwargs,
-        )
-    if pushed:
-        for stage in view.stages:
-            for out in stage.outs:
-                workspace, _key = out.index_key
-                _update_out_meta(out, repo.index.data[workspace])
-            stage.dump(with_files=True, update_pipeline=False)
+    for remote_name, view in worktree_view_by_remotes(
+        repo.index, push=True, targets=targets, **kwargs
+    ):
+        remote_obj = _get_remote(repo, remote_name, remote, "push")
+        new_index = view.data["repo"]
+        if remote_obj.worktree:
+            logger.debug("indexing latest worktree for '%s'", remote_obj.path)
+            old_index = build_data_index(view, remote_obj.path, remote_obj.fs)
+            logger.debug("Pushing worktree changes to '%s'", remote_obj.path)
+        else:
+            old_index = None
+            logger.debug(
+                "Pushing version-aware files to '%s'", remote_obj.path
+            )
+
+        if remote_obj.worktree:
+            diff_kwargs: Dict[str, Any] = {
+                "meta_only": True,
+                "meta_cmp_key": partial(_meta_checksum, remote_obj.fs),
+            }
+        else:
+            diff_kwargs = {}
+
+        total = len(new_index)
+        with Callback.as_tqdm_callback(
+            unit="file",
+            desc=f"Pushing to remote {remote_obj.name!r}",
+            disable=total == 0,
+        ) as cb:
+            cb.set_size(total)
+            pushed += checkout(
+                new_index,
+                remote_obj.path,
+                remote_obj.fs,
+                old=old_index,
+                delete=remote_obj.worktree,
+                callback=cb,
+                latest_only=remote_obj.worktree,
+                jobs=jobs,
+                **diff_kwargs,
+            )
+
+        for out in view.outs:
```

Comment:
@skshetry This actually fixes #8760 for `push_worktree`. 
Could you include the fix for `update_worktree_stages` in this P.R?

Reply:
Would that be okay if I do after this PR? I don't want to touch `update_worktree_stages` in this PR.

Reply:
> Would that be okay if I do after this PR? I don't want to touch update_worktree_stages in this PR.

Ok, I will do it after this one is merged

Reply:
Looks like I missed that `dvc update` works differently with cloud versioning. 😞 
So I need to split them in multiple views based on remotes they are going to.